print: Support different output formats

stephen-fox opened the pull request:
The `-print` argument now accepts the following values:

- `raw` - Write the chunk's value as-is to stdout
- `decompress` - Decompress the chunk's value prior to writing it to stdout
- `decompress-hexdecode` - Decompress and hex-decode the chunk's value prior to writing it to stdout. ImageMagick appears to first hex-encode and then compress the data using zlib compression

stephen-fox committed:
png: Include offset in TextChunk type.

stephen-fox committed:
print: Support different output formats.

jnschaeffer commented on README.md:
Typo

stephen-fox replied:
D'oh x1!

jnschaeffer commented on README.md:
Typo

stephen-fox replied:
D'oh x2!

jnschaeffer commented on README.md:
Maybe it's worth adding a warning in here and the help text highlighting that this could be dangerous?

stephen-fox committed:
main: Fixed function Go doc formatting.

Whoever decided code block indentation needs to be made go fmt's
responsibility... I hate you.

jnschaeffer commented on main.go:
```diff
@@ -80,20 +83,126 @@ func main() {
-		os.Exit(1)
+	for _, chunk := range chunks {
+		if chunk.Keyword == profileKeyword {
+			log.Printf("***POTENENTIAL INDICATOR OF COMPROMPISE*** - "+
```
Typo

stephen-fox replied:
D'oh x3!

jnschaeffer commented on main.go:
```diff
@@ -80,20 +83,126 @@ func main() {
+func writeChunkTo(chunk *png.TextChunk, writer io.Writer) error {
+	reader := io.NopCloser(bytes.NewReader(chunk.Value))
+
+	if strings.HasPrefix(printContents, "decompress") && chunk.Compressed {
```
Might be better to make the allowed values constants and then check to see if `printContents` is one of the values that involves decompression.

stephen-fox replied:
oki

stephen-fox committed:
general: Fixed embarrassing typos.

stephen-fox committed:
main: Move output modes to global constants.

stephen-fox committed:
readme: Really emphasize the dangers.

stephen-fox committed:
main: Forgot to use output consts in flag usage.
